ambari-server/src/main/java/org/apache/ambari/server/api/resources/ResourceInstanceFactoryImpl.java

```diff
@@ -254,6 +254,10 @@ public static ResourceDefinition getResourceDefinition(Resource.Type type, Map<R
         resourceDefinition = new ExtensionLinkResourceDefinition();
         break;
 
+      case DefaultOperatingSystem:
+        resourceDefinition = new DefaultOperatingSystemResourceDefinition();
```

Comment:
You can use `org.apache.ambari.server.api.resources.SimpleResourceDefinition` here instead:

```
resourceDefinition = new SimpleResourceDefinition(Resource.Type.DefaultOperatingSystem, "default_operating_system", "default_operating_systems");
```

Reply:
Oh yeah, forgot about this one. Let me switch it. Thanks!